view/frontend/web/js/view/pdp/pdp-fast-button.js

```diff
@@ -1,23 +1,19 @@
-define(['uiComponent', 'jquery', 'ko', 'underscore', 'fastConfig'],
-    function(Component, $, ko, _, fastConfigFactory) {
+define(['fastButtonBase', 'jquery', 'ko', 'underscore'],
+    function(Component, $, ko, _) {
         'use strict';
 
-        var fastConfig = fastConfigFactory();
-
         return Component.extend({
 
             initialize: function() {
                 var self = this;
                 this._super();
-                self.shouldShowFastButton = ko.observable(fastConfig.shouldShowFastOnPDP());
+                self.shouldShowFastButton = ko.observable(self.shouldShowFastButton());
                 $(document).ready(function () {
-                    $("#pdp-fast-button").css({
-                        'width': ($("#product-addtocart-button").outerWidth() + 'px')
-                    });
                     $("#pdp-fast-button").prependTo(".box-tocart .fieldset .actions");
```

Comment:
I'm also wondering if this line can be removed or not. this is also css change using jQuery and is not best practice  for knockoutjs. Basically remove all lines starting line [#11 to #13 ](https://github.com/fast-af/magento2/blob/f83a7672b7614ae680c6c1dfa85b8ade2309b3fe/view/frontend/web/js/view/pdp/pdp-fast-button.js#L11-L13)

Reply:
The `prependTo` is there to position the button where we want it. I think it's necessary because there is no other way to control where the button is located using Magento's theming tools. I agree it would be nice to remove it, but I'm not sure if we can.

Reply:
For the CSS override that we removed (previous lines 14-16), Jenn brought up a good point that there is no easy way to style the button with CSS because you don't know the width of the add-to-cart button. For sites that don't have dev teams, we need to have a way to cleanly control how the button is styled.

My thought for that would be to add a setting so that we can conditionally load that part. Something like 

```
if (fastConfig.setButtonWidth) {
    $("#pdp-fast-button").css({
        'width': ($("#product-addtocart-button").outerWidth() + 'px')
    });
}
```

Reply:
I see, strange that we cannot control where fast button is inserted in magento.
One option to expose additional config to set width for fast checkout button and then use css binding on the template like [this](https://knockoutjs.com/documentation/style-binding.html). But still not going to work if we cant insert where we need. 
sounds like we are not able to avoid css override

Reply:
I'll look into the style binding. Ideal situation is to be able to set it without code (or have a good default setting) but make it possible for a dev team to override the style for specific sites.

I think the placement issue is a limitation in how Magento add-to-cart forms are rendered.